components/ProfileMenu.vue

```diff
@@ -270,16 +304,65 @@ const joinedContentLanguages = computed(() =>
               <path d="M4 15H20" stroke="currentColor" stroke-width="2" />
             </svg>
           </button>
+          <div>
+            {{ t("profile.preference-language-" + Math.min(i + 1, 4), i + 1) }}
+          </div>
+          <div
+            class="text-label-1 bg-background-1 min-w-min px-2 py-1.5 rounded-lg font-semibold shadow-[0_4px_12px_0_#0000000D,0_1px_4px_0_#0000000D,0_0_0_1px_#0000000D]"
+          >
+            <select v-model="contentLanguages[i]" class="py-1 w-full">
+              <option
+                v-for="(lang, i) in availableContentLanguages
+                  .filter((x) => !contentLanguages.includes(x))
+                  .concat([item])"
+                :key="`Lang${i}`"
+                :value="lang"
+              >
+                {{
+                  `${languageDictionary[lang]?.NativeName} (${languageDictionary[lang]?.EnglishName})`
+                }}
+              </option>
+            </select>
+          </div>
+
           <div
+            v-if="false"
             class="text-black-1 bg-background-1 dark:bg-background-3 dark:text-black-1 min-w-[100px] pl-3 py-2.5 shadow ring-1 ring-label-separator rounded-lg"
           >
-            {{ lang }}
+            {{ item }}
           </div>
-          <button v-if="i > 0" class="text-2xl">
+          <button
+            v-if="i > 0"
+            class="text-2xl"
+            @click="
+              () => {
+                contentLanguages.splice(i, 1);
+              }
+            "
+          >
             <NuxtIcon name="icon.close.small" />
           </button>
         </div>
       </VueDraggable>
+      <div
+        v-if="
+          availableContentLanguages.filter(
+            (x) => !contentLanguages.includes(x),
+          )[0]
+        "
+        class="text-label-3 p-3 flex flex-row gap-2 mt-4"
+        @click="
```

Comment:
slimmed it a bit.
But it's pretty annoying. It should just be contentLanguages.push(nextUnusedContentLanguage()), but it's complaining that it might be undefined even though the v-if already checks for that.